### PR TITLE
fixed softmax not to use remap() to avoid memory copy

### PR DESCRIPTION
**Pull request by YuyaKobayashi**
At least DenseASPP(SOFTMAX with 4 classes) and JointFace worked.

Transpose of input is not tested.

### CODE REVIEW

File: common/src/dmp_network.cpp [hunk 1]
```diff
@@ -327,51 +340,39 @@ static void run_softmax_2(__fp16* dst, __fp16* src, int* dim, int dim_size) {
...
-    src_buffer = dst;
-  }
-
   switch (softmax_axis) {
     case 0:
       run_softmax_0(reinterpret_cast<__fp16*>(dst_buffer), 
                     reinterpret_cast<__fp16*>(src_buffer),
-                    layer.input_dim, layer.input_dim_size);
+                    layer.input_dim, layer.input_dim_size,
+					          layer.is_input_hw_layout, need_transpose);
```
Comment (ZongHong-Lyu) [inline common/src/dmp_network.cpp]: Also please replace the tabs here with spaces, and also the following two blocks

**Comment by YuyaKobayashi** [comment]
I modified to avoid `need_transpose`/`is_input_hw_layout` check

File: common/src/dmp_network.cpp [hunk 2]
```diff
@@ -236,26 +237,50 @@ static void float_to_fp16(__fp16* __restrict dst, const float* __restrict src, i
...
 
-#define _IDX(x, y, c) ((((x) * (H)) + (y)) * (C) + (c))
-static void run_softmax_0(__fp16* dst, __fp16* src, int* dim, int dim_size) {
+static int _src_idx_softmax(int x, int y, int c, int W, int H, int C) {
+  int block_base = C & ~0x7;
+  int block_idx = C >> 3;
+  int cinb = c - block_base;
+  int clen_block = c >= block_base ? C - block_base + 1: 8;
+  return block_idx * 8 * W * H + (y * W + x) * (clen_block * W * H) + cinb;
+}
```
Comment (ZongHong-Lyu) [inline common/src/dmp_network.cpp]: The calculation here seems to be wrong.
should be:
```cpp
  int clen_block = c >= block_base ? C - block_base : 8;
  return block_idx * 8 * W * H + (y * W + x) * clen_block + cinb;
```
You can see the `remap` function for the correct calculation too.

Comment (YuyaKobayashi) [reply common/src/dmp_network.cpp]: Exactly

**Pull request merged**